persist/fs/msgpack/stream.go

```diff
@@ -59,11 +59,23 @@ type DecoderStream interface {
 
 	// Remaining returns the remaining bytes in the stream.
 	Remaining() int64
+
+	// Offset returns the current offset in the byte stream
+	Offset() int
 }
 
+// DecoderStream is a data stream that is read by the decoder,
+// it takes both a reader and the underlying backing bytes.
+// This is constructed this way since the decoder needs access
+// to the backing bytes when taking refs directly for decoding byte
+// slices without allocating bytes itself but also needs to progress
+// the reader (for instance when a reader is a ReaderWithDigest that
+// is calculating a digest as its being read).
```

Comment:
Does the comment need to be duplicated here considering it's already on the interface?

Reply:
nope I meant to delete that, thanks